### ISSUE TITLE
Crashes when saving SSP models on a partition that does not contain `/tmp`

### ISSUE THREAD

**Issue by pstelzig** [OpenModelica/OpenModelica#9426]
### Description

On Ubuntu 22.04, there is a bug apparently caused by boost 1.74 which is shipped with Ubuntu 22.04. **This bug appears only if your /tmp directory is on another partition than where you save your OpenModelica files**, e.g. your home partition. 

This is a common setup, where the home folder lives on a dedicated home partition (often preferred by more advanced linux users), whereas the tmp folder lives on another partition.

### Steps to Reproduce

**Assume, your OMEdit working directory is on your home folder, and this is on another partition than the standard `/tmp` directory.** 

1. Create a new empty SSP model
2. Save it to your home folder as `somecosim.ssp`
 
OMEdit will crash with somehting like (see `/tmp/OpenModelica_<your user name>/OMEdit/omeditoutput.txt`)

```
terminate called after throwing an instance of 'boost::filesystem::filesystem_error'
  what():  boost::filesystem::copy_file: Invalid cross-device link: "/tmp/OpenModelica_<your user name>/OMEdit/somecosim-akai5asi/temp/model.ssp", "/home/<your user name>/<your working dir>/somecosim.ssp"
```

### Expected Behavior

Save SSP model successfully.

### Version and OS

 - OpenModelica Version: 1.19.2
 - OS: Ubuntu 22.04
 
### Additional Context: Workaround

To avoid this situation, I forced OpenModelica to use a tmp folder that is on the same partition where I keep my files. I did this by starting `OMEdit` with `TMPDIR=<some path on my home directory>`, e.g. my OpenModelica working directory: 

```
TMPDIR=$HOME/<my OpenModelica working dir> OMEdit
```

For convenience, I added this to my `$HOME/.bashrc`

```bash
alias OMEdit='TMPDIR=$HOME/<my OpenModelica working dir> OMEdit'
```

Now, every time I call `OMEdit` from the terminal, the tmp directory used by OMEdit will be created in my OpenModelica working directory.

I prefer to have a launcher item on my gnome deskop. So I created a file `$HOME/.local/share/applications/omedit_loc_tmp.desktop` with the following content: 

```
[Desktop Entry]
Version=1.0
Encoding=UTF-8
Name=OMEdit local tmp
GenericName=Physical modeling and simulation tool
Comment=A Modelica connection editor
TryExec=OMEdit
Exec=env TMPDIR=/home/<my user name>/<my OpenModelica working dir> OMEdit %f
Icon=/usr/share/pixmaps/omedit.png
Terminal=false
Type=Application
StartupNotify=true
Categories=Development;Education;Science;Math;
MimeType=text/x-onb;application/x-omnotebook;
```

which is basically identical to `/usr/share/applications/omedit.desktop` but with the `Exec` field modified as above. After logging our and back in, this launcher will appear in the gnome application search alongside the original launcher item from the OpenModelica installation.

**Comment by casella** [comment]
This is weird, why should the installation device be relevant when accessing a Linux file system?

**Comment by pstelzig** [comment]
In virtual machines I don't use a separate partition from home. There I did not have the issues as described above. 

On my workstations on the other hand, where I use this setting with dedicated partitions for root and home, I observed this issue. 

Weird indeed...

**Comment by casella** [comment]
Adding our Linux guru @sjoelund. Any insight?

**Comment by sjoelund** [comment]
Yes, it's a bug with Boost: https://github.com/boostorg/filesystem/issues/254... But why isn't std::filesystem used instead? It's supposed to be as the OMSimulator code uses it. Anyway, this is a bug for the OMSimulator repository and would be best fixed by making sure boost isn't used on a modern OS.

**Comment by casella** [comment]
@arun3688 do you have a chance of looking at this?

**Comment by arun3688** [comment]
@casella I don't have linux installed in my laptop

**Comment by adrpo** [comment]
We should change the code to use std::filesystem instead of boost if possible.

**Comment by arun3688** [comment]
@adrpo do we support `c++17`  ?

**Comment by arun3688** [comment]
@pstelzig this should be fixed with this commit f92b7f01cc8cdbb34ee24b005fccc50d4d7579db, since we moved to oms3, all the export and import of ssp is handled via the python library, so this should be automatically fixed please test it from nightly builds and reopen the issue in case if issue exist
